algebird-core/src/main/scala/com/twitter/algebird/Field.scala

```diff
@@ -54,6 +54,14 @@ object FloatField extends Field[Float] {
     assertNotZero(r)
     l / r
   }
+  override def sum(t: TraversableOnce[Float]): Float = {
+    var sum = 0.0f
+    t.foreach(sum += _)
```

Comment:
did we find out in the CMS PR that foreach doesn't compile down to a while?
should this be a while loop? I could imagine dynamic dispatch on a closure costingmore than boxing, but who knows